VocabularyTrainer/Home Screen/CollectionViewCell.swift

```diff
@@ -0,0 +1,119 @@
+import UIKit
+
+class CollectionViewCell: UICollectionViewCell {
+
+    typealias Dimensions = LanguageCellViewModel.Dimensions
+    typealias Colors = LanguageCellViewModel.Colors
+
+    private let titleLabel: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.textColor = HomeViewModel.Colors.title
+        label.font = .preferredFont(forTextStyle: .headline)
+        label.numberOfLines = 0
+        return label
+    }()
+
+    private let subtitleLabel: UILabel = {
+        let label = UILabel()
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.textColor = HomeViewModel.Colors.subtitle
+        label.font = .preferredFont(forTextStyle: .body)
+        label.numberOfLines = 0
+        return label
+    }()
+
+    private let labelContainer: UIStackView = {
+        let stackView = UIStackView()
+        stackView.translatesAutoresizingMaskIntoConstraints = false
+        stackView.axis = .vertical
+        stackView.spacing = Dimensions.labelSpacing
+        stackView.distribution = .fill
+        return stackView
+    }()
+
+    /// Image view showing the language's icon / image / emoji.
+    private let imageView: UIImageView = {
+        let view = UIImageView()
+        view.translatesAutoresizingMaskIntoConstraints = false
+        view.contentMode = .scaleAspectFit
+        view.image = UIImage(systemName: "hare")
+        view.clipsToBounds = true
+        return view
+    }()
+
+    private let emojiLabel: UILabel = {
+        let label = UILabel()
+        label.numberOfLines = 1
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.font = .preferredFont(forTextStyle: .headline)
+        return label
+    }()
+
+    private let background: UIView = {
+        let view = UIView()
+        view.backgroundColor = Colors.cellBackground
+        view.layer.cornerRadius = Dimensions.cornerRadius
+        return view
+    }()
+
+    private let selectedBackground: UIView = {
+        let view = UIView()
+        view.backgroundColor = Colors.selectedCellBackground
+        view.layer.cornerRadius = Dimensions.cornerRadius
+        return view
+    }()
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        setUp()
+    }
+
+    /// Not supported.
+    required init?(coder: NSCoder) { nil }
+
+    func configure(with item: LanguageCellViewModel) {
+        titleLabel.text = item.languageName
+        subtitleLabel.text = item.subtitle
+        emojiLabel.text = item.emoji
+    }
+
+    private func setUp() {
+
+        backgroundView = background
+        selectedBackgroundView = selectedBackground
+        contentView.backgroundColor = .clear
+
+        labelContainer.addArrangedSubview(titleLabel)
+        labelContainer.addArrangedSubview(subtitleLabel)
+        contentView.addSubview(labelContainer)
+//        contentView.addSubview(imageView)
+        contentView.addSubview(emojiLabel)
```

Comment:
Maybe we could split in two methods setUpUI() and setUpConstraints(), what do you think?

Reply:
good idea! updated that: 54d50af1aa2377f5023e3721449aa1fd1c1b9645